recipes/self-managed-otlp-ingest/README.md

```diff
@@ -0,0 +1,150 @@
+```
+
+This script creates a properly configured Collector with relevant permissions & recommended settings.
+
+> [!IMPORTANT]
+> The following IAM permissions are required to be configured if you have enabled GKE Workload Identity Federation. This example uses GKE Autopilot, which has Workload Identity Federation enabled by default.
+> It is necessary to configure Workload Identity to let the applications running in this cluster authenticate to Google Cloud APIs.
+
+For clusters with GKE Workload Identity Federation enabled, grant the deployed collector relevant IAM permissions:
+```shell
+# This is the number corresponding to your project ID, you can get this by visiting the Google Cloud Platform home page.
+export PROJECT_NUMBER=<your-project-number>
+
+gcloud projects add-iam-policy-binding projects/$PROJECT_ID \
+    --role=roles/logging.logWriter \
+    --member=principal://iam.googleapis.com/projects/$PROJECT_NUMBER/locations/global/workloadIdentityPools/$PROJECT_ID.svc.id.goog/subject/ns/opentelemetry/sa/opentelemetry-collector \
+    --condition=None
+gcloud projects add-iam-policy-binding projects/$PROJECT_ID \
+    --role=roles/monitoring.metricWriter \
+    --member=principal://iam.googleapis.com/projects/$PROJECT_NUMBER/locations/global/workloadIdentityPools/$PROJECT_ID.svc.id.goog/subject/ns/opentelemetry/sa/opentelemetry-collector \
+    --condition=None
+gcloud projects add-iam-policy-binding projects/$PROJECT_ID \
+    --role=roles/cloudtrace.agent \
+    --member=principal://iam.googleapis.com/projects/$PROJECT_NUMBER/locations/global/workloadIdentityPools/$PROJECT_ID.svc.id.goog/subject/ns/opentelemetry/sa/opentelemetry-collector \
+    --condition=None
+```
+
+At this point, there is an OpenTelemetry Collector running in the cluster and is ready to receive telemetry.
+
+### Injecting language specific auto-instrumentation
+
+Using the OpenTelemetry Operator, we can inject language specific auto-instrumentation in applications deployed in the cluster.\
+This is done by first creating the Instrumentation CRD (Custom Resource Definition) and then using annotations to apply the instrumentation to a specific Kubernetes resource.
+
+```shell
+# Create the Instrumentation CRD
+kubectl apply -f instrumentation.yaml
+```
+
+Next, annotate the deployment in which the created instrumentation needs to be injected:
+
+```shell
+# Annotate the deployment to inject the created instrumentation
+# This will trigger a rolling restart of the deployment
+kubectl patch deployment.apps/quickstart-app -p '{"spec":{"template":{"metadata":{"annotations":{"instrumentation.opentelemetry.io/inject-java": "sample-java-auto-instrumentation"}}}}}'
+```
+
+## Viewing the result
+
+After injecting the application, the application should start producing telemetry which can be viewed in Google Cloud.\
+The instrumentation used in the sample is currently configured to generate only traces and the traces can be viewed on the GCP UI using the Trace Explorer.
```

Comment:
Why only traces?  Is this just as a starting point?

Reply:
Added metrics.